### PR TITLE
Cross-version API Compatibility tests - CLI Discovery sources APIs

### PR DESCRIPTION
**Pull request by mpanchajanya**
### What this PR does / why we need it

- Added Config CLI Discovery sources APIs compatibility tests  for SetCLIDiscoverySource, GetCLIDiscoverySource, DeleteCLIDiscoverySource APIs for Runtime libraries latest, v0.28.0
- Added helper methods to create API Commands, Validators to validate input and outputOptions.
- Extended the testplugins implementation to trigger CLI Discovery sources APIs.
- Added Default helper functions to create the input and output options data for command creation

Detailed Info:

`test/compatibility`
`core/api_constants.go` -> Added CLI Discovery sources APIs names

`framework` changes
`compatibility-tests/clidiscoverysources` -> Add compatibility tests for CLI discovery sources APIs

`clidiscoverysources` directory to provide functions to create api commands and validations to help write compatibility tests forCLI discovery sources APIs

`testplugins` changes
Add Implementation to trigger CLI discovery sources APIs for all supported runtime versions

### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR

- Added Unit testing for the helper functions like Command creations, Trigger Feature APIs.
- Build and Run compatibility tests using below command and make sure all tests pass
`make compatibility-tests`

```
Ginkgo ran 6 suites in 25.350487917s
Test Suite Passed


```

### Release note

```release-note
Added CLI discovery sources config APIs compatibility tests
```

### CODE REVIEW

**Comment by vuil** [comment]
@mpanchajanya changes looks good in general, but these class of changes continue to be hard to review.

qq: any reason why we do not need to use a more specific validation strategy?

on a related note: (this is more for the earlier PR 65): we support providing customized strategy for the env APIs but I don't see it getting used. Was that an oversight?

**Comment by mpanchajanya** [comment]
> @mpanchajanya changes looks good in general, but these class of changes continue to be hard to review.
> 
> qq: any reason why we do not need to use a more specific validation strategy?
> 
> on a related note: (this is more for the earlier PR 65): we support providing customized strategy for the env APIs but I don't see it getting used. Was that an oversight?

@vuil  Thanks for finding this. I have updated tests to use validation strategy for discovery sources, env and feature flags

**Review by vuil — APPROVED** [review]
thanks for the updates, lgtm

File: test/compatibility/framework/clidiscoverysources/cli_discoverysources_validators.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+package clidiscoverysources
+
+import (
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/tanzu-plugin-runtime/test/compatibility/core"
+)
+
+// Validate  the setDiscoverySourceInputOptions as per runtime version i.e. check whether mandatory fields are set and throw error if missing
```
[the remark this answers is not in the document]
Comment (mpanchajanya) [reply test/compatibility/framework/clidiscoverysources/cli_discoverysources_validators.go:12]: removed extra space

**Pull request merged**